x-pack/plugin/async-search/build.gradle

```diff
@@ -13,6 +13,7 @@ base {
 addQaCheckDependencies(project)
 
 dependencies {
+  testImplementation project(':server')
```

Comment:
I tried to move (rather than copy) ThrowingQueryBuilder from
``
./x-pack/plugin/async-search/src/test/java/org/elasticsearch/xpack/search/ThrowingQueryBuilder.java
```
to 
```
./server/src/test/java/org/elasticsearch/search/query/ThrowingQueryBuilder.java
```
in order to use it for both CrossClusterSearchIT and CrossClusterAsyncSearchIT,
and I tried adding this entry so that `CrossClusterAsyncSearchIT` could import
ThrowingQueryBuilder from server/src/test, but it didn't work. I'm not sure how 
get that working so I just copied the file.

Reply:
@mark-vieira - is this something you could help me with?

Reply:
If you want to use _test_ code from `:server` then you need the following:
```groovy
testImplementation testArtifact(project(':server'))
```

Reply:
Thank you! That worked.